### ISSUE TITLE
lua api way to check if a player exist

### ISSUE THREAD

**Issue by volkj** [minetest/minetest#4942]
Currently minetest.get_player_by_name( playername ) gives a player object only if there's a player, as an online player; it doesn't work for offline players, so it can't be used to check if a player already exists.

A function like minetest.player_exists( playername ) can be done checking core.auth_table for a playername key:

    function minetest.player_exists( playername )
        if core.auth_table[ playername ] then
            return true
        end
        return nil
    end

**Comment by adrido** [comment]
Please don't use the minetest.auth_table or core.auth_table.
Depending on the authentication handler this table does not exist, or is not filled correctly.
For example a authentication handler that stores the users in a mysql database instead of a file.
More Info:
http://dev.minetest.net/minetest.register_authentication_handler

**Comment by SmallJoker** [comment]
So, instead it would be ...?
```Lua
function minetest.player_exists(playername)
	return minetest.builtin_auth_handler.get_auth(playername) ~= nil
end
```

**Comment by sofar** [comment]
Why would player presence be dependent on authentication protocol used? What if, in the future, it could support 2 auth protocols at the same time?

I'd rather have a mechanism that works irregardless of the auth mechanism..

**Comment by adrido** [comment]
Not shure if I did understand you correctly, but afaik the auth handler has nothing to do with the auth-protocoll. Minetest asks the auth-handler if there is a account `get_auth(name)` then it does either check the pwhash and log the player in, or call the auth_handler's `create_auth(name, pwhash) `
if the player changes the password Minetest calls the set_password(player, pw) of the auth handler.
If the privs of the player changes Minetest calls the set_privs(player, privs) of the auth handler.
Why should there be two different auth handler? Sorry If I did not get the point or did you understand wrong.

@SmallJoker the correct code would be:
```
function minetest.player_exists(playername)
	return minetest.get_auth_handler().get_auth(playername) ~= nil
end
```
get_auth_handler returns either the builtin_auth_handler, or the auth_hander that got registered with register_auth_handler.
https://github.com/minetest/minetest/blob/0.4.15/builtin/game/auth.lua#L177
